modules/refine-model/src/main/java/org/openrefine/model/changes/ChangeData.java

```diff
@@ -20,6 +20,13 @@
  */
 public interface ChangeData<T> extends CloseableIterable<IndexedData<T>> {
 
+    /**
+     * A marker at the end of the serialization of a partition of a change data, signalling that the end of the
+     * partition was reached. This is useful when iterating synchronously from a ChangeData object which is being
+     * written simultaneously.
+     */
+    public static final String partitionEndMarker = "end";
```

Comment:
This string gets written at the end of change data partition files to signal that they are complete. That's a change in project serialization format (but of course given that the new architecture has only been released as alpha, we can do that quite freely).